testing/dart/compositing_test.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data' show Float64List;
+import 'dart:ui';
+
+import 'package:test/test.dart';
+
+void main() {
+  test('pushTransform throws with invalid matrx', () {
+    final SceneBuilder builder = SceneBuilder();
+    final Float64List matrix4 = Float64List(16);
+    matrix4[0] = double.nan;
+    expect(
+      () => builder.pushTransform(matrix4),
+      throwsA(TypeMatcher<ArgumentError>())
```

Comment:
nit: `const TypeMatcher<ArgumentError>` and below

Reply:
Done in the new version. I just realized that this comment was made before I uploaded the new patch...